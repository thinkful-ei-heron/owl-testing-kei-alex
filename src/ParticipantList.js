import React from 'react';
import Participant from './Participant';
import './ParticipantList.css';

export default function ParticipantList (props) {
  let participantsSort = props.participants;
  
  participantsSort.sort(function(a, b) {
    let textA=a.name;
    let textB=b.name;
    return (textA < textB) ? -1 : (textA > textB) ? 1 : 0;
  })
  let absent=participantsSort.filter(element => element.inSession===false);
  let inSession=participantsSort.filter(element => element.inSession===true && element.onStage===false);
  let onStage=participantsSort.filter(element => element.inSession===true && element.onStage===true);
  return (
  <ul className='participant-list'>
    {onStage.map((element, index) => 
    <li key={index} className='user-tab'><Participant 
      avatar={element.avatar}
      name={element.name}
      inSession={element.inSession}
      onStage={element.onStage}
    /></li>)}
    {inSession.map((element, index) => 
    <li key={index} className='user-tab'><Participant 
      avatar={element.avatar}
      name={element.name}
      inSession={element.inSession}
      onStage={element.onStage}
    /></li>)}
    {absent.map((element, index) => 
    <li key={index} className='user-tab'><Participant 
      avatar={element.avatar}
      name={element.name}
      inSession={element.inSession}
      onStage={element.onStage}
    /></li>)}
  </ul>
  )
}