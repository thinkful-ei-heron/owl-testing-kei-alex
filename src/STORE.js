const STORE = {
  participants: [
    {
      id: 1,
      name: 'Koren Templeton',
      avatar:
          'https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1',
      inSession: true,
      onStage: true,
      user: false
    },
    {
      id: 2,
      name: 'Caty Flucker',
      avatar:
          'https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1',
      inSession: true,
      onStage: true,
      user: true
    },
    {
      id: 3,
      name: 'Axe Kubicka',
      avatar: 'https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1',
      inSession: false,
      onStage: false,
      user: false
    },
    {
      id: 4,
      name: 'Frank Runciman',
      avatar:
          'https://robohash.org/etexercitationemassumenda.jpg?size=200x200&set=set1',
      inSession: true,
      onStage: false,
      user: false
    },
    {
      id: 5,
      name: 'Ashla Attwool',
      avatar:
          'https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1',
      inSession: true,
      onStage: true,
      user: false
    }
  ]
};

export default STORE;