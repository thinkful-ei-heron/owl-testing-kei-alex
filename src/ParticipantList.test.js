import React from 'react';
import ReactDOM from 'react-dom';
import STORE from './STORE';
import ParticipantList from './ParticipantList';
import renderer from 'react-test-renderer';

describe('ParticipantList tests', () => {
  //smoke test
  it('render with prop', () => {
    const div = document.createElement('div');
    ReactDOM.render(<ParticipantList
      participants = {STORE.participants}
    />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  //snapshot test
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<ParticipantList
        participants = {STORE.participants}
      />)
      .toJSON();
      expect(tree).toMatchSnapshot();
  })
});