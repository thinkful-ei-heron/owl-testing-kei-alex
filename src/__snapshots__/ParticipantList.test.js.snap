// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParticipantList tests renders the UI as expected 1`] = `
<ul
  className="participant-list"
>
  <li
    className="user-tab"
  >
    <div
      className="Participant"
    >
      <img
        alt=""
        className="user-avatar"
        src="https://robohash.org/iustodoloremqueinventore.jpg?size=200x200&set=set1"
      />
      <h3
        className="user-name"
      >
        Ashla Attwool
      </h3>
      <span
        className="dot in-session"
      />
      <span
        className="status"
      >
        on stage
      </span>
    </div>
  </li>
  <li
    className="user-tab"
  >
    <div
      className="Participant"
    >
      <img
        alt=""
        className="user-avatar"
        src="https://robohash.org/doloredolorescupiditate.jpg?size=200x200&set=set1"
      />
      <h3
        className="user-name"
      >
        Caty Flucker
      </h3>
      <span
        className="dot in-session"
      />
      <span
        className="status"
      >
        in session
      </span>
    </div>
  </li>
  <li
    className="user-tab"
  >
    <div
      className="Participant"
    >
      <img
        alt=""
        className="user-avatar"
        src="https://robohash.org/etexercitationemassumenda.jpg?size=200x200&set=set1"
      />
      <h3
        className="user-name"
      >
        Frank Runciman
      </h3>
      <span
        className="dot in-session"
      />
      <span
        className="status"
      >
        in session
      </span>
    </div>
  </li>
  <li
    className="user-tab"
  >
    <div
      className="Participant"
    >
      <img
        alt=""
        className="user-avatar"
        src="https://robohash.org/itaquevoluptatumrerum.jpg?size=200x200&set=set1"
      />
      <h3
        className="user-name"
      >
        Koren Templeton
      </h3>
      <span
        className="dot in-session"
      />
      <span
        className="status"
      >
        on stage
      </span>
    </div>
  </li>
  <li
    className="user-tab"
  >
    <div
      className="Participant"
    >
      <img
        alt=""
        className="user-avatar"
        src="https://robohash.org/utlaborumfugit.jpg?size=200x200&set=set1"
      />
      <h3
        className="user-name"
      >
        Axe Kubicka
      </h3>
      <span
        className="dot hidden"
      />
      <span
        className="status"
      >
        in session
      </span>
    </div>
  </li>
</ul>
`;
